core/domain/exp_jobs_one_off_test.py

```diff
@@ -1834,3 +1840,298 @@ def test_no_action_is_performed_for_deleted_exploration(self):
 
         run_job_for_deleted_exp(
             self, exp_jobs_one_off.TranslatorToVoiceArtistOneOffJob)
+
+
+class VoiceoverDurationSecondsOneOffJobTests(test_utils.GenericTestBase):
+    ONE_OFF_JOB_MANAGERS_FOR_TESTS = [
+        exp_jobs_one_off.VoiceoverDurationSecondsOneOffJob]
+
+    VALID_EXP_ID = 'exp_id0'
+
+    USERNAME_A = 'usernamea'
+    EMAIL_A = 'emaila@example.com'
+    TEST_AUDIO_FILE_MP3 = 'cafe.mp3'
+    AUDIO_UPLOAD_URL_PREFIX = '/createhandler/audioupload'
+    _FILENAME_PREFIX = 'audio'
+
+    def setUp(self):
+        super(VoiceoverDurationSecondsOneOffJobTests, self).setUp()
+        self.signup(self.EMAIL_A, self.USERNAME_A)
+
+        self.user_a_id = self.get_user_id_from_email(self.EMAIL_A)
+        self.process_and_flush_pending_tasks()
+
+    def test_voiceover_duration_secs_job_fails_with_invalid_exploration(self):
+        observed_log_messages = []
+
+        def _mock_logging_function(msg, *args):
+            """Mocks logging.error()."""
+            observed_log_messages.append(msg % args)
+
+
+        exploration = exp_domain.Exploration.create_default_exploration(
```

Comment:
Here and below, only have a maximum of one blank line at a time within a method.

Reply:
Ok

Reply:
done.